### PR TITLE
techmino: init at 0.17.16

### PR DESCRIPTION
**Pull request by chayleaf**
###### Description of changes

https://github.com/26F-Studio/Techmino (a Love2D Tetris clone). It uses a custom [native Lua module](https://github.com/26F-Studio/cold_clear_ai_love2d_wrapper) for its AI, and that module is a C library that links to [a Rust library](https://github.com/26F-Studio/cold-clear/tree/lua-api)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by oluceps** [comment]
Result of `nixpkgs-review pr 225445` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>techmino</li>
  </ul>
</details>

**Review by oluceps — APPROVED** [review]

File: pkgs/by-name/te/techmino/package.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{ lib
+, stdenv
+, fetchurl
+, callPackage
+, makeWrapper
+, makeDesktopItem
+, love
+, luajit
```
Comment (pbsds) [inline pkgs/by-name/te/techmino/package.nix:8]:
```suggestion
, luajit
, libcoldclear ? callPackage ./libcoldclear.nix { }
, ccloader ? callPackage ./ccloader.nix { inherit libcoldclear luajit; }
```

File: pkgs/by-name/te/techmino/package.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      hash = "sha256-+j+8m2vwaWgHYSFL6urvTcB0vA+PCZ+FYJ22CNXfcSc=";
+    };
+    comment = description;
+    desktopName = "Techmino";
+    genericName = "Tetris Clone";
+    categories = [ "Game" ];
+  };
+
+  libcoldclear = callPackage ./libcoldclear.nix { };
+  ccloader = callPackage ./ccloader.nix { inherit libcoldclear luajit; };
```
Comment (pbsds) [inline pkgs/by-name/te/techmino/package.nix:30]:
```suggestion
```

File: pkgs/by-name/te/techmino/package.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    mkdir -p $out/bin
+    makeWrapper ${love}/bin/love $out/bin/techmino \
+      --add-flags $out/share/games/lovegames/techmino.love \
+      --suffix LUA_CPATH : ${ccloader}/lib/lua/${luajit.luaversion}/CCLoader.so
+
+    mkdir -p $out/share/applications
+    ln -s ${desktopItem}/share/applications/* $out/share/applications/
+  '';
+
```
Comment (pbsds) [inline pkgs/by-name/te/techmino/package.nix:59]:
```suggestion

  passthru = {
    inherit libcoldclear ccloader;
  };

```

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 225445` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>techmino</li>
  </ul>
</details>

Plays fine

Let's make the sub packages accessible and overridable

**Comment by chayleaf** [comment]
also made it run pre/post phase hooks

**Pull request merged**